File: 0101-symmetric-tree/0101-symmetric-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
typedef struct TreeNode node;
bool find(node*root1,node*root2){
    if(root1==NULL && root2==NULL)return true;
    if(root1==NULL && root2!=NULL)return false;
    if(root1!=NULL && root2==NULL)return false;
    if(root1->val!=root2->val)return false;
    bool l1=find(root1->left,root2->right);
    bool l2=find(root1->right,root2->left);
    return l1 && l2;
}
bool isSymmetric(struct TreeNode* root) {
    return find(root->left,root->right);
}

File: 0101-symmetric-tree/0101-symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean find(TreeNode root1,TreeNode root2){
        if(root1 == null && root2 == null)return true;

        if(root1 != null && root2 != null){

            // System.out.println(root1.val+" "+root2.val);

            if(root1.val != root2.val)return false;

            if(find(root1.right,root2.left) && find(root1.left,root2.right))return true;

            return false;
        }

        return false;
        
    }
    public boolean isSymmetric(TreeNode root) {
        return find(root.left,root.right);
    }
}